cmd/root_test.go

```diff
@@ -2141,6 +2142,79 @@ func TestProjectRootIsSymlink(t *testing.T) {
 	)
 }
 
+func TestConcurrentInvocation(t *testing.T) {
+	as := require.New(t)
+
+	tempDir := test.TempExamples(t)
+	configPath := filepath.Join(tempDir, "/treefmt.toml")
+
+	test.ChangeWorkDir(t, tempDir)
+
+	cfg := &config.Config{
+		FormatterConfigs: map[string]*config.Formatter{
+			"echo": {
+				Command:  "echo",
+				Includes: []string{"*"},
+			},
+			"slow": {
+				Command: "test-fmt-delayed-append",
+				// connect timeout for the db is 1 second
+				// wait 2 seconds before appending ' ' to each provided path
+				Options:  []string{"2", " "},
+				Includes: []string{"*"},
+			},
+		},
+	}
+
+	eg := errgroup.Group{}
+
+	// concurrent invocation with one slow instance and one not
+
+	eg.Go(func() error {
+		treefmt(t,
+			withArgs("--formatters", "slow"),
+			withConfig(configPath, cfg),
+			withNoError(t),
+		)
+
+		return nil
+	})
+
+	time.Sleep(time.Second)
```

Comment:
Why the sleep? Is it to give the previous treefmt a moment to start up (and lock the boldb cache) before we start the next treefmt?

Am I understanding this correctly?

- At time 0: We start a treefmt ("slow") that will grab the bolddb cache, and sleep 2 seconds before formatting and releasing the bolddb cache.
- Sleep 1 second
- At time 1: We start a second treefmt ("echo") that will wait up to 1 second to grab the boltdb cache.
- At time 2: "slow" is finishing up while "echo" is just about ready to give up on waiting for the cache. 

IIUC, this feels racy to me, it's not a sure thing that "echo" will always hit the "failed to open cache" error. I'd fix this by just making "slow" quite a bit slower (5 or 10 seconds feels like plenty to me).

Reply:
I thought I had reduced that to 500ms. Made the change.